### PR TITLE
Mct4039

### PR DESCRIPTION
**Pull request by scottbell**
Fixes https://github.com/nasa/openmct/issues/4039

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/plot/stackedPlot/StackedPlot.vue [hunk 1]
```diff
@@ -159,9 +160,9 @@ export default {
 
         exportJPG() {
             this.hideExportButtons = true;
-            const plotElement = this.$refs.plotContainer;
+            const plotElement = this.$el;
```
Comment (scottbell) [inline src/plugins/plot/stackedPlot/StackedPlot.vue:163]: plotContainer wasn't resolving, so switched to just $el. there's probably a more elegant way to do this

File: src/plugins/plot/stackedPlot/StackedPlot.vue [hunk 2]
```diff
@@ -170,9 +171,9 @@ export default {
         exportPNG() {
             this.hideExportButtons = true;
 
-            const plotElement = this.$refs.plotContainer;
+            const plotElement = this.$el;
```
Comment (scottbell) [inline src/plugins/plot/stackedPlot/StackedPlot.vue:174]: same as above

File: src/plugins/plot/stackedPlot/StackedPlot.vue (hunk 1 above)
Comment (scottbell) [reply src/plugins/plot/stackedPlot/StackedPlot.vue:163]: this was preventing StackedPlots from exporting images by the way. this seemed broken in master as well.

**Review by akhenry — CHANGES_REQUESTED** [review]
This looks great, thanks! 

My only suggestion would be that we might want to add a test that exercises the `renderElement` function. ie. create a dom element, invoke `renderElement` and pass in the element, then check that the promise that is returned resolves with a blob. I don't think we can really do much beyond checking that it returns a blob. Trying to glean anything meaningful from the blob itself is probably a waste of time.

File: src/exporters/ImageExporter.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+    return filename.replace(regex, '_');
+}
+
+/**
+ * canvas.toBlob() not supported in IE < 10, Opera, and Safari. This polyfill
+ * implements the method in browsers that would not otherwise support it.
+ * https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob
+ */
+function polyfillToBlob() {
```
[the remark this answers is not in the document]
Comment (scottbell) [reply src/exporters/ImageExporter.js]: Polyfill is toast

**Review by akhenry — DISMISSED** [review]

**Review by akhenry — APPROVED** [review]

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**